lib/shoryuken/polling.rb

```diff
@@ -102,5 +123,94 @@ def queue_weight(queues, queue)
         queues.count { |q| q == queue }
       end
     end
+
+    class StrictPriority < BaseStrategy
+
+      def initialize(queues)
+        # Mapping of queues to priority values
+        @queue_priorities = queues
+          .each_with_object(Hash.new(0)) { |queue, h| h[queue] += 1 }
+
+        # Priority ordering of the queues
+        @queue_order = @queue_priorities
+          .to_a
+          .sort_by { |queue, priority| -priority }
+          .map(&:first)
+
+        # Pause status of the queues
+        @queue_status = @queue_order
+          .each_with_object({}) { |queue, h| h[queue] = [true, nil] }
+
+        # Most recently used queue
+        @current_queue = nil
```

Comment:
From what  I understand, that it actually `last_used_queue`. Can we name it like that? It will make more sense for the loop.